### PR TITLE
Support repo-config v0.9 properly

### PR DESCRIPTION
**Pull request by tiyash-basu-frequenz**
This repo recently moved to repo-config v0.9. While the dependabot PR for the upgrade passed, the release notes for repo-config suggest more changes. This commit tries to add these additional requirements for repo-config v0.9 upgrade.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -7,34 +7,44 @@
 # https://github.com/marketplace/actions/labeler
 
 "part:docs":
```
Comment (Marenz) [inline .github/labeler.yml:9]: While this change is good and important, it requires you to also use labeler ver. 5: https://github.com/frequenz-floss/frequenz-api-microgrid/blob/3ad22440c2039207de4e901ec465fe4ab756b04d/.github/workflows/labeler.yml#L21

Comment (Marenz) [reply .github/labeler.yml:9]: See also https://github.com/frequenz-floss/frequenz-api-microgrid/pull/190

Comment (Marenz) [reply .github/labeler.yml:9]: (you probably can just cherry-pick that commit from the PR)

**Review by Marenz — COMMENTED** [review]
Without the labeler update this will break labeling after merge.
Also I notice you didn't add the fix to have git-support in pyproject for the test-installation ci job. 
Which isn't a problem as long as git-urls are not used as repository dependencies..
Here is the patch in question:

```
patch -p1 <<'EOF'
diff --git a/.github/containers/test-installation/Dockerfile b/.github/containers/test-installation/Dockerfile
index 772b2ae..2494545 100644
--- a/.github/containers/test-installation/Dockerfile
+++ b/.github/containers/test-installation/Dockerfile
@@ -6,7 +6,12 @@

 FROM --platform=${TARGETPLATFORM} python:3.11-slim

-RUN python -m pip install --upgrade --no-cache-dir pip
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y \
+    git && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    python -m pip install --upgrade --no-cache-dir pip

 COPY dist dist
 RUN pip install dist/*.whl && \
EOF
```

File: .github/labeler.yml (hunk 1 above)
Comment (tiyash-basu-frequenz) [reply .github/labeler.yml:9]: Done :rocket:

**Comment by tiyash-basu-frequenz** [comment]
>Also I notice you didn't add the fix to have git-support in pyproject for the test-installation ci job.

It is because there is no test-installation step for this repo (which is by intention), so this step does not apply here.

**Review by Marenz — APPROVED** [review]

**Review by thea-leake — APPROVED** [review]
LGTM!

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
         # only use hashes to pick the action to execute (instead of tags or branches).
         # For more details read:
         # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
-        uses: actions/labeler@ac9175f8a1f3625fd0d4fb234536d26811351594  # 4.3.0
+        uses: actions/labeler@v5.0.0
```
Comment (llucax) [inline .github/workflows/labeler.yml:21]: We should use the hash here for security reasons.

Comment (llucax) [reply .github/workflows/labeler.yml:21]: * https://github.com/frequenz-floss/frequenz-api-microgrid/pull/230

**Pull request merged**